rollouts/api/v1alpha1/rollout_types.go

```diff
@@ -157,6 +161,24 @@ type GitLabSelector struct {
 	Branch string `json:"branch,omitempty"`
 }
 
+// OCISource defines configuration to discover OCI packages.
+type OCISource struct {
+	// image is the OCI image repository URL for the package to sync from.
+	// e.g. `LOCATION-docker.pkg.dev/PROJECT_ID/REPOSITORY_NAME/PACKAGE_NAME`.
+	// The image can be pulled by TAG or by DIGEST if it is specified in PACKAGE_NAME.
+	// - Pull by tag: `LOCATION-docker.pkg.dev/PROJECT_ID/REPOSITORY_NAME/PACKAGE_NAME:TAG`.
+	// - Pull by digest: `LOCATION-docker.pkg.dev/PROJECT_ID/REPOSITORY_NAME/PACKAGE_NAME@sha256:DIGEST`.
+	// If neither TAG nor DIGEST is specified, it pulls with the `latest` tag by default.
+	// Required
+	Image string `json:"image"`
+
+	// dir is the absolute path of the directory that contains
+	// the local resources.  Default: the root directory of the image.
+	// Note (droot): We will extend `Dir` to express variants of a package at some point.
```

Comment:
So this doesn't yet take into account variants right? Similar to with git, there are probably several ways for users to organize packages and variants across one or more images, so I would expect we eventually need something similar to the selector pattern from the github/gitlab sources. Do we think we can add this later without needing backwards incompatible changes to the API?

Reply:
Though the current schema is extensible by supporting glob patterns for `image` and `dir` fields, but the naming was inconsistent with the other package sources `github` and `gitlab`. Changed the schema to add `OCISelector` field which is now consistent with other selectors. We can extend `image` and `dir` to support glob patterns in the future to support variants (that will also require changes to package-matcher component as well).